### PR TITLE
Add all describers to the exported DefaultObjectDescriber

### PR DESCRIPTION
**Pull request by vihangm**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
The kubectl describe package is mainly used by the kubectl describe command,
and as such the describer implementations are not exported.
The exported Describer interfaces require a `clientset.Interface` to be
provided and reach out to the API to fetch the runtime object before
describing it.

However this package also exports `DefaultObjectDescriber` which is
particularly useful if you already have an in-memory `runtime.Object`
that one would like to describe. This makes it easier to describe objects
passed to Informer event handlers.

The `DefaultObjectDescriber` seems to be missing many of the commonly
used k8s objects and this PR adds the remaning unexported describer
implementations to the object to allow one to describe in-memory
Deployments, Statefulsets etc.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 1]
```diff
@@ -381,14 +381,44 @@ var DefaultObjectDescriber ObjectDescriber
 func init() {
 	d := &Describers{}
 	err := d.Add(
+		describeNamespace,
```
Comment (mpuckett159) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:384]: Can we sort this list of describers alphabetically?

**Review by soltysh — APPROVED** [review]
/triage accepted
/priority backlog
/approve

ping either myself or @mpuckett159 for final tag once you sort the methods alphabetically

**Comment by mpuckett159** [comment]
/lgtm

Thank you so much! Maybe a good follow on could be to investigate why not all of the describe functions return the expected signature and if viable we could do a similar conformance effort to the splitting of options and flags for commands, and try and get all those synced up. First step would be to create an issue describing this and we can investigate further.

**Pull request merged**